core/src/main/java/org/jdbi/v3/core/argument/EnumArgumentFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.argument;
+
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.internal.EnumStrategy;
+import org.jdbi.v3.core.qualifier.QualifiedType;
+
+class EnumArgumentFactory implements QualifiedArgumentFactory {
```

Comment:
Note how this QualifiedArgumentFactory replaces all three implementations below, and delegates only the bits that actually vary to the `EnumStrategy`.